src/Storage/Controllers/MessageboxInstancesController.cs

```diff
@@ -109,7 +109,7 @@ public async Task<ActionResult> SearchMessageBoxInstances([FromBody] MessageBoxQ
 
                 if (queryResponse?.Exception != null)
                 {
-                    return StatusCode(500, queryResponse.Exception);
+                    return StatusCode(cancellationToken.IsCancellationRequested ? 499 : 500, queryResponse.Exception);
```

Comment:
How can you be sure that the cause of the exception is a client side timeout? I think you need to check to cancelllationToken also.